File: baml_language/sdk_tests/crates/swift/function_calls/customizable/TestUnhandledSpawnErrors.swift
import Foundation
import XCTest
import Baml
import BamlBridge

final class TestUnhandledSpawnErrors: XCTestCase {
    func test_unhandled_spawn_error_uses_host_default() throws {
        if ProcessInfo.processInfo.environment["BAML_SWIFT_UNHANDLED_SPAWN_CHILD"] != nil {
            XCTAssertEqual(try Baml.spawn_unhandled_error(), 1)
            BamlRuntime.shared.shutdown()
            XCTFail("unhandled spawn error did not terminate the process")
            return
        }

        let process = Process()
        process.executableURL = URL(fileURLWithPath: "/usr/bin/xcrun")
        process.arguments = [
            "xctest",
            "-XCTest",
            "TestUnhandledSpawnErrors/test_unhandled_spawn_error_uses_host_default",
            Bundle(for: Self.self).bundleURL.path,
        ]
        process.environment = ProcessInfo.processInfo.environment.merging(
            ["BAML_SWIFT_UNHANDLED_SPAWN_CHILD": "1"]
        ) { _, child in child }
        let stderr = Pipe()
        process.standardError = stderr

        try process.run()
        process.waitUntilExit()

        let output = String(
            decoding: stderr.fileHandleForReading.readDataToEndOfFile(),
            as: UTF8.self
        )
        XCTAssertNotEqual(process.terminationStatus, 0, output)
        XCTAssertTrue(output.contains("user.unhandled_spawn_error"), output)
    }
}

File: baml_language/sdk_tests/crates/java/function_calls/customizable/TestUnhandledSpawnErrors.java
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

import baml_bridge.BamlFfi;
import baml_sdk.Fns;
import java.nio.charset.StandardCharsets;
import org.junit.jupiter.api.Test;

class TestUnhandledSpawnErrors {
    @Test
    void test_unhandled_spawn_error_uses_host_default() throws Exception {
        String javaBin = System.getProperty("java.home") + "/bin/java";
        String classpath = System.getProperty("java.class.path");
        Process process =
                new ProcessBuilder(
                                javaBin,
                                "-cp",
                                classpath,
                                "TestUnhandledSpawnErrors$UnhandledSpawnSnippet")
                        .redirectErrorStream(true)
                        .start();

        String output = new String(process.getInputStream().readAllBytes(), StandardCharsets.UTF_8);
        assertEquals(0, process.waitFor(), output);
        assertTrue(output.contains("user.unhandled_spawn_error"), output);
    }

    public static final class UnhandledSpawnSnippet {
        public static void main(String[] args) {
            assertEquals(1L, Fns.spawn_unhandled_error());
            BamlFfi.shutdownRuntime();
        }
    }
}
